core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -2265,8 +2265,8 @@ private void assertTypeAndToString(
     checkSimplify(coalesce(gt(nullInt, nullInt), trueLiteral),
         "true");
     checkSimplify(coalesce(unaryPlus(nullInt), unaryPlus(vInt())),
-        "+(?0.int0)");
-    checkSimplifyUnchanged(coalesce(unaryPlus(vInt(1)), unaryPlus(vInt())));
+        "?0.int0");
```

Comment:
Should we add more test cases which covers decimal as well?

Reply:
Sounds good. I have added some test cases for decimal. Please take a look.